### DISCUSSION TITLE
Modifying Mask Tool to allow custom image rotation/aspect ratio/

### DISCUSSION THREAD

**Discussion by crypticsymmetry** [sn4k3/UVtools#954] [Scripts]
Hi can someone help me add image rotation functionality to the built in mask tool  , or custom script that does the same thing as the mask tool but with image rotation to allow using different aspect ratio images, as currently it will squeeze the image if its a landscape image and works fine for portrait aspect ratio as seen below. 

![PhotoPaper_exposure_with_Resin_printer_Mask](https://github.com/user-attachments/assets/9bd2a78e-cd01-4cfb-ad8d-523d1d3223d6)
![PhotoPaper_exposure_with_Resin_printer](https://github.com/user-attachments/assets/7189b25b-8734-4236-a682-06726338f1b6)

im currently using the mask to to expose cyanotypes. and would like to be able to change how the rotation gets applied to the image/layers.

UVtools\UVtools.Core\Operations\OperationMask.cs
UVtools\UVtools.UI\Controls\Tools\ToolMaskControl.axaml
UVtools\UVtools.UI\Controls\Tools\ToolMaskControl.axaml.cs

Thanks!

**Comment by sn4k3** [comment]
I think you might using the wrong tool? Give a peak at Litophane tool, it might be what you want, and have all that options.

**Reply by crypticsymmetry** [reply]
ive tried the litho tool but the mask tool works better as it uses the whole grayscale image every layer instead of separating the grayscale values over all the layers, like each layer has a different "shade", where i want the whole grayscale image exposed every layer, as its exposing on a piece of paper like these
![202aef21-7bfd-49a1-8e49-bb8dd760fd31](https://github.com/user-attachments/assets/43502bcc-b117-4dbe-962e-dfb23233c1d4)

**Reply by sn4k3** [reply]
As mask is designed to mask current pixels with your image it may lose original pixel from your image.
I would add a new option in litophane to allow expose whole image instead of separating, would that work for you?

**Reply by crypticsymmetry** [reply]
that sound perfect. thanks for the quick replies

**Reply by sn4k3** [reply]
You can try it with the latest version. Report back with the result.

**Reply by crypticsymmetry** [reply]
Thanks that works perfect!
